src/lib.rs

```diff
@@ -514,16 +609,39 @@ struct FaceProperties {
 /// Either a raw binary data or a file path.
 ///
 /// Stores the whole font and not just a single face.
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub enum Source {
     /// A font's raw data. Owned by the database.
     Binary(Vec<u8>),
 
     /// A font's path.
     #[cfg(feature = "fs")]
     File(PathBuf),
+
+    /// A font's raw data originating from a shared file mapping.
+    #[cfg(all(feature = "fs", feature = "memmap"))]
+    SharedFile(PathBuf, Arc<dyn AsRef<[u8]> + Sync + Send>),
+
+    /// A font's raw data. Shared between the database and the application.
+    SharedBinary(Arc<Vec<u8>>),
```

Comment:
Besides what this patch proposes, there are two other options for changing this enum:

1. Change `Binary(Vec<u8>)` to `Binary(Arc<Vec<u8>>)`. That eliminates an extra vector copying when calling `make_shared_face_data` when the corresponding face ID was previously initialised by a supplied `Vec`. However that's source incompatible :-(

2. Combine `SharedBinary` and `SharedFile` into one `SharedBinary(Option<PathBuf>, Arc<dyn AsRef<[u8]> + Sync + Send>)` variant. That minimises the API surface, but it makes the enum variant harder to use (marginally, one can argue).

I'd be happy to change to either of those two variants or any other proposed version, of course.

Reply:
Yes, 4 variants is too much. `fontdb` must be simple first. 
You can change `Binary` to store `Arc`, but then we have to remove `Arc<Source>` from everywhere, because it's no longer needed.

Reply:
Sounds good, I'll try that.